src/EditorFeatures/Core/SymbolSearch/SymbolSourceProvider.cs

```diff
@@ -0,0 +1,26 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.ComponentModel.Composition;
+using Microsoft.VisualStudio.Language.Intellisense.SymbolSearch;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.SymbolSearch
+{
+    [Export(typeof(ISymbolSourceProvider))]
+    [ContentType(ContentTypeNames.RoslynContentType)]
+    [Name("Roslyn symbol source")]
+    class SymbolSourceProvider : ISymbolSourceProvider
+    {
+        ImmutableArray<ISymbolSource> ExportedSources;
```

Comment:
why not just make this readonly and initialize in the constructor?  because laszy seems to save a single alloc.

Reply:
we're passing in "this" to have access to MEF parts in the actual implementation.